iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -135,9 +135,25 @@ Viewer.prototype._bindHandlers = function () {
     $('#iframe-container .zoom-in').click(function () { viewer.zoomIn() });
     $('#iframe-container .zoom-out').click(function () { viewer.zoomOut() });
 
+    // Listen to messages posted to this window
+    $(window).on("message", function(e) { viewer.handleMessage(e) });
```

Comment:
Perhaps `$(window).on("message", viewer.handleMessage);`?

Reply:
I had tried that already and it doesn't work, as `this` is not populated in `handleMessage` when invoked that way, and so felt out of scope for this change.

Reply:
Oops, ok.  I don't do JavaScript often enough to remember the nuances of `this`.

Reply:
I suspect this is also the reason why Paul handle the zoom in/out click handling using anonymous lambda functions above.